esp-hal-common/src/aes/mod.rs

```diff
@@ -0,0 +1,195 @@
+//! Advanced Encryption Standard (AES) support.
+//!
+//! This module provides functions and structs for AES encryption and
+//! decryption.
+//!
+//! ### Features
+//! The AES peripheral has the following features available on individual chips:
+//!
+//! | Feature          | ESP32    | ESP32-C3 | ESP32-S2 | ESP32-S3 |
+//! |------------------|----------|----------|----------|----------|
+//! |AES128            |Y         |Y         |Y         |Y         |
+//! |AES192            |Y         |N         |Y         |N         |
+//! |AES256            |Y         |Y         |Y         |Y         |
+//! |Custom endianness |Y         |N         |Y         |N         |
+//!
+//! ### Implementation State
+//! * DMA mode is currently not supported.
+
+use core::marker::PhantomData;
+
+use crate::{
+    peripheral::{Peripheral, PeripheralRef},
+    peripherals::{
+        generic::{Readable, Reg, RegisterSpec, Resettable, Writable},
+        AES,
+    },
+    system::PeripheralClockControl,
+};
+
+#[cfg_attr(esp32, path = "esp32.rs")]
+#[cfg_attr(esp32s3, path = "esp32_s3.rs")]
+#[cfg_attr(esp32s2, path = "esp32_s2.rs")]
+#[cfg_attr(esp32c3, path = "esp32_c3.rs")]
```

Comment:
```suggestion
#[cfg_attr(esp32s3, path = "esp32s3.rs")]
#[cfg_attr(esp32s2, path = "esp32s2.rs")]
#[cfg_attr(esp32c3, path = "esp32c3.rs")]
```

Sorry for this, but we do not use underscores in other files like these in the HAL (`esfuse/esp32xx.rs`, `clock/clocks_ll/esp32xx.rs`, etc.), so would you mind please changing these?